### ISSUE TITLE
🐛 BUG: `astro preview` stopped working

### ISSUE THREAD

**Issue by herteleo** [withastro/astro#2981]
### What version of `astro` are you using?

0.26.1

### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

`npm run preview` starts without error but shows `ERR_CONNECTION_REFUSED` if you try to open `http://localhost:3000/` in the browser.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-uy2pjg?file=package.json

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by SebasG22** [comment]
+ This is failing on 1.0.0-beta.0

**Comment by SlicedSilver** [comment]
I'm also experiencing this bug (also on a mac). 

What appears to be happening is that the `preview` command runs without any errors and immediately quits back to the terminal. I tried running the command with the `--verbose` flag and with the debug env variables `DEBUG=astro:*,vite:* astro preview` but nothing is reported.

**Comment by SlicedSilver** [comment]
https://github.com/withastro/astro/blob/50bd14aca8a67512b18d4e9981d43007623e20b8/packages/astro/src/cli/index.ts#L139-L145
I've noticed that the `preview` function above is immediately resolving (on my setup)  and thus not staying open. I wasn't able to see any recent changes that explain why this behaviour has changed. 
https://github.com/withastro/astro/blob/50bd14aca8a67512b18d4e9981d43007623e20b8/packages/astro/src/core/preview/index.ts#L109

---

A naive  solution seems to be changing the code to resemble the `dev` function by adding a `return await new Promise(() => {});`. Not sure what the knock-on effects of this change would be, but it does seem to resolve the issue on my device. 
```javascript
case "preview": {
      try {
        await preview(config, { logging });
        return await new Promise(() => {
        });
      } catch (err) {
        return throwAndExit(err);
      }
    }
```

Hopefully this helps to track down the root cause.

**Comment by idrakimuhamad** [comment]
Running `preview` in `1.0.0-beta.2` will display below and the task stopped. I've tried running `build` first and the `preview` but its just the same.

```sh
🚀  astro  v1.0.0-beta.2 started in 2ms
  
  ┃ Local    http://localhost:3000/
  ┃ Network  use --host to expose
  ```

**Comment by rutzfutz** [comment]
The same error happens on Windows 10 for me.

**Comment by karthik-srinivas** [comment]
yes this is happening for me as well 

<img width="487" alt="image" src="https://user-images.githubusercontent.com/45588403/161834628-d1dc25f6-5a87-421b-9463-2eebacf23ee2.png">

the process is getting terminated just after running `preview`  in my case `serve`

**Comment by tony-sull** [comment]
Thanks for all the debugging and details in this issue!  Just merged #3004 with a fix
